src/components/_classes/component/Component.js

```diff
@@ -3388,14 +3387,15 @@ export default class Component extends Element {
     if (flags.silentCheck) {
       return [];
     }
+    let dirty = this.dirty || flags.dirty
     if (this.options.alwaysDirty) {
-      flags.dirty = true;
+      dirty = true;
     }
-    if (flags.fromSubmission && this.hasValue(data)) {
-      flags.dirty = this.pristine && this.component.protected ? false : true;
+    if (flags.fromSubmission && !_.isUndefined(_.get(flags?.submission?.data, this.key)) && !(this.pristine && this.protected)) {
+      dirty = this.pristine && this.component.protected ? false : true;
```

Comment:
It seems to me that this.protected is always undefined. I was not able to find a place where it is set to true/false.
Why do we need the check for 'protected' and 'pristine' in both line 3395 and line 3394?